uwblueprint/focus-on-nature
# feat: batch process camper cancellations

## Notion ticket link

[Cancellation/Refund: Auto cancel camper registration and send cancellation confirmation email to parent on successful refund](https://www.notion.so/uwblueprintexecs/Cancellation-Refund-Auto-cancel-camper-registration-and-send-cancellation-confirmation-email-to-par-52223aabb0b9464ea6645e4affbe5149?pvs=4)



## Implementation description
* Implemented batch processing camper cancellations
* Note this does not have its own Notion ticket as it's part of a larger ticket



## Steps to test
Note: deletion currently can't be processed unless payment is actually made through Stripe

1. In Postman, send a DELETE request to http://localhost:3000/campers/cancel-registration, giving `chargeId` as a key and array of `camperId`s (see below for reference)
2. Test that cancellation does not go through if session start date is less than 30 days and waitlist is empty
3. Test that cancellation does not go through for session start date greater than 30 days no payment was made

<img width="1112" alt="Screen Shot 2023-06-19 at 9 49 07 PM" src="https://github.com/uwblueprint/focus-on-nature/assets/88808428/755899d9-9c2a-4e2c-a70f-904648022ccf">



## What should reviewers focus on?
When payment hasn't been made:
* Check outputs correct error message

When payment has been made:
* Check refund amount is correct
* Check campers are deleted after their cancellations have been processed


## Checklist
